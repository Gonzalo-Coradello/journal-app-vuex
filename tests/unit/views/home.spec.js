import HomeView from '@/views/HomeView.vue'
import { shallowMount } from '@vue/test-utils'
import { describe, expect, test, vi } from 'vitest'

describe('Pruebas en el Home View', () => {
  test('debe renderizar el componente correctamente', () => {
    const wrapper = shallowMount(HomeView)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('hacer click en un botÃ³n debe redireccionar a no-entry', () => {
    const mockRouter = {
      push: vi.fn()
    }

    const wrapper = shallowMount(HomeView, {
      global: {
        mocks: {
          $router: mockRouter
        }
      }
    })

    wrapper.find('button').trigger('click')

    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
  })
})
