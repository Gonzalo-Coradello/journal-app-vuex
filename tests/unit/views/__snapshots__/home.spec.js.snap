// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en el Home View > debe renderizar el componente correctamente 1`] = `
"<main>
  <h1>Hola mundo</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</main>"
`;
